src/test/java/com/target/studentinfo/controller/StudentControllerTest.java

```diff
@@ -0,0 +1,131 @@
+package com.target.studentinfo.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.target.studentinfo.dto.request.StudentRequest;
+import com.target.studentinfo.exception.BadRequestException;
+import com.target.studentinfo.exception.ErrorCode;
+import com.target.studentinfo.model.Student;
+import com.target.studentinfo.service.StudentService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(StudentController.class)
+@ComponentScan(basePackages = "com.target.studentinfo")
+public class StudentControllerTest {
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private StudentService studentService;
+    Student student;
+    private List<Student> mockStudents;
+    @BeforeEach
+    void setup(){
+        mockStudents = Arrays.asList(
+                new Student(1L, "Ram", "k", "ram@123.com", "A+", "Ashok", "Swathi", 12, "male", "hyderabad", 8, "painting", "none", "self", false),
+                new Student(2L, "John", "Deo", "john@123.com", "B+", "Smith", "Jane", 14, "male", "kerala", 10, "music", "dust", "self", true),
+                new Student(3L, "Raksh", "k", "raksh@123.com", "A+", "Ashok", "Swathi", 10, "male", "hyderabad", 6, "painting", "none", "self", false)
+        );
+        student=Student.builder()
+                .id(1L)
+                .firstName("Ram")
+                .lastName("K")
+                .emailId("ram@123.com")
+                .bloodGroup("A+")
+                .fatherName("Ashok")
+                .motherName("Swathi")
+                .age(12)
+                .gender("male")
+                .address("Hyderabad")
+                .standard(8)
+                .extraCurricular("painting")
+                .allergies("none")
+                .transport("self")
+                .isActive(true).build();
```

Comment:
You can define student1, student2 and student3 and add them to `mockStudents` list.